lib/plugins/aws/package/compile/events/apiGateway/lib/stage/index.js

```diff
@@ -133,11 +133,18 @@ module.exports = {
   },
 };
 
-function getLogGroupResource(service, stage) {
-  return {
+function getLogGroupResource(service, stage, provider) {
+  const resource = {
     Type: 'AWS::Logs::LogGroup',
     Properties: {
       LogGroupName: `/aws/api-gateway/${service}-${stage}`,
     },
   };
+
+  const logRetentionInDays = provider.getLogRetentionInDays();
+  if (logRetentionInDays !== null) {
```

Comment:
I would simplify that to `if (logRetentionInDays)`, as technically best contract is to treat any falsy value as no _logRetentionInDays_.

Then if for some reason due refactoring, we will suddenly start returning `undefined` in `getLogRetentionInDays` it should not bite us back.

Reply:
Updated. It slipped my mind that 0 wasn't a valid value here.